src/wallet/rpcnames.cpp

```diff
@@ -823,6 +825,151 @@ name_update ()
 
 /* ************************************************************************** */
 
+RPCHelpMan
+queuerawtransaction ()
+{
+  return RPCHelpMan ("queuerawtransaction",
+      "\nQueue a transaction for future broadcast.",
+      {
+          {"hexstring", RPCArg::Type::STR, RPCArg::Optional::NO, "The hex string of the raw transaction"},
+      },
+      RPCResult {RPCResult::Type::STR_HEX, "", "the transaction ID"},
+      RPCExamples {
+          HelpExampleCli("queuerawtransaction", "txhex") +
+          HelpExampleRpc("queuerawtransaction", "txhex")
+      },
+      [&] (const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
+{
+  std::shared_ptr<CWallet> const wallet = GetWalletForJSONRPCRequest (request);
+  if (!wallet) return NullUniValue;
+  CWallet* const pwallet = wallet.get ();
+
+  RPCTypeCheck (request.params,
+                {UniValue::VSTR});
+
+  // parse transaction from parameter
+  CMutableTransaction mtxParsed;
+  if (!DecodeHexTx(mtxParsed, request.params[0].get_str(), true, true))
+    throw JSONRPCError(RPC_DESERIALIZATION_ERROR, "TX decode failed");
+  CTransactionRef txParsed(MakeTransactionRef(mtxParsed));
+  const uint256& hashTx = txParsed->GetHash();
+
+  // Validate transaction
+  NodeContext& node = EnsureAnyNodeContext(request.context);
+  ChainstateManager& chainman = EnsureChainman(node);
+  CTxMemPool& mempool = EnsureMemPool(node);
+
+  LOCK2 (pwallet->cs_wallet, cs_main);
```

Comment:
Do you need to hold both locks at the same time?  Wouldn't it be possible to just hold `cs_main` while you try the mempool accept, and then hold just the wallet lock when you queue the transaction?

Reply:
Done.